### DISCUSSION TITLE
Cloning repository via ssh avatar and username not showing

### DISCUSSION THREAD

**Discussion by kleninmaxim** [community/community#45048] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

When I cloned the repository via ssh and push to the github repository, I don't see my avatar and my username.
Image:
![image](https://user-images.githubusercontent.com/54952996/213905372-a4cb5d28-8787-462c-b6bd-9241455c2704.png)

When I clone an https repository and push with my github token, the avatar and username are shown in the repository.
Image:
![image](https://user-images.githubusercontent.com/54952996/213905391-9f80f1e7-2af1-447a-9263-ee82135c0b58.png)

What can I do to show the avatar and username via ssh as well?

**Comment by jge162** [comment, marked as answer]
@kleninmaxim remember when you clone a repository using SSH, you are not providing your GitHub credentials with the push. In order to show your avatar and username when pushing via SSH, you will need to configure your SSH key with your GitHub account.

set it up like this...
```
a. Generate an SSH key on your local machine if you don't have one already.
b. Go to your GitHub settings and select "SSH and GPG keys"
c. Click "New SSH key" and give it a title (e.g. "My local machine")
d. Paste in the contents of your public SSH key and save.
e. On your local machine, add the private key to your SSH agent:

ssh-add ~/.ssh/id_rsa
```
Now, when you clone the repository via ssh and push, your avatar and username should be associated with the push.

**Reply by kleninmaxim** [reply]
> @kleninmaxim remember when you clone a repository using SSH, you are not providing your GitHub credentials with the push. In order to show your avatar and username when pushing via SSH, you will need to configure your SSH key with your GitHub account.
> 
> set it up like this...
> 
> ```
> a. Generate an SSH key on your local machine if you don't have one already.
> b. Go to your GitHub settings and select "SSH and GPG keys"
> c. Click "New SSH key" and give it a title (e.g. "My local machine")
> d. Paste in the contents of your public SSH key and save.
> e. On your local machine, add the private key to your SSH agent:
> 
> ssh-add ~/.ssh/id_rsa
> ```
> 
> Now, when you clone the repository via ssh and push, your avatar and username should be associated with the push.

I did the following and still the avatar is not displayed on commits and pushes.

1. Command: ` ssh-keygen -t ed25519 -C "your_email@example.com"`, `your_email@example.com` change on my
2. Command: `eval "$(ssh-agent -s)"` and get: `Agent pid 12039`
3. Command: `ssh-add ~/.ssh/id_ed25519`
4. Copied content from `~/.ssh/id_ed25519.pub`
5. Go to GitHub settings, select "SSH and GPG keys", then click "New SSH key" and give it a title and paste in the contents of my public SSH key and save.
6. Command: `ssh -T git@github.com` and get: `Hi kleninmaxim! You've successfully authenticated, but GitHub does not provide shell access.`
7. Then I cloned repository to local machine, made commit and push it.

And my avatar and username also not showing.

**Reply by jge162** [reply]
@kleninmaxim check your Git configurations, make sure that your Git configuration is set up correctly with your username and email. 

```You can check your current configuration by running the command "git config --list" in the terminal. If your name and email are not set, you can set them with the following commands: "git config --global user.name 'Your Name' " and "git config --global user.email '[your_email@example.com](mailto:your_email@example.com)'".```


```The avatar will only show up on the commit if the email address that you used to make the commit matches the email address associated with your GitHub account.```

make sure the repository, owners can choose to not display avatars or usernames of contributors on the commit, so you should check the settings of the repository you are pushing to.

I doubt it, but sometimes the cache of your browser or the browser you are using to check the commits may hold the old information. Try to clear the cache or use another browser to check the commits.

this should fix your issue.

**Reply by kleninmaxim** [reply]
Thank you! I had to specify the mail in the settings.
`git config --global --replace-all user.email 'your_email@example.com'`

**Reply by jge162** [reply]
Perfect 👍 np

**Reply by Marcosaurios** [reply]
Indeed I was missing the step:

`f. Update your local git config to use the same email and username as the one registered with the SSH key`.
